main.rb

```diff
@@ -0,0 +1,122 @@
+# Enumerable
+module Enumerable
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    return to_enum(__method__) unless block_given?
+
+    if is_a?(Array)
+      size.times do |i|
+        yield self[i]
+      end
+    else size.times do |i|
+      yield [keys[i], values[i]]
+    end
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(__method__) unless block_given?
+
+    size.times do |i|
+      yield self[i], i
+    end
+  end
+
+  def my_select
+    return to_enum(__method__) unless block_given?
+
+    array = []
+    my_each do |i|
+      array << i if yield i
+    end
+    array
+  end
+
+  def my_all?(*args)
+    if block_given?
+      my_each { |i| return false unless yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return false unless i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return false if (args[0] =~ i).nil? }
+    else
+      my_each { |i| return false unless i.nil? }
+    end
+    true
+  end
+
+  def my_any?(*args)
+    if block_given?
+      my_each { |i| return true if yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return true if i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return true unless (args[0] =~ i).nil? }
+    else
+      my_each { |i| return true if i.nil? }
+    end
+    false
+  end
+
+  def my_none?(*args)
+    if block_given?
+      my_each { |i| return false if yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return false if i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return false unless (args[0] =~ i).nil? }
+    else
+      my_each { |i| return false if i == true }
+    end
+    true
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
[nil, false, nil, 5].my_none? == [nil, false, nil, 5].none? => false
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/85830957-b3c95300-b74a-11ea-9375-2d4e08b434e9.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690365-cd4ed900-6f68-11ea-9933-f73b0091208d.png)

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[3,4,7,11].my_none?(3) == [3,4,7,11].none?(3) => false
```
- Yours is retrieving and error
![image](https://user-images.githubusercontent.com/52209067/85831154-14589000-b74b-11ea-9767-cb0acdfde5bd.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690915-a9d85e00-6f69-11ea-81b3-7adb07b281ca.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end

To make this work we have to use block.call(element) instead of yield(element) in my_any?